std/file.d

```diff
@@ -5119,3 +5120,77 @@ string tempDir() @trusted
     myFile.write("hello");
     assert(myFile.readText == "hello");
 }
+
+/**
+* Returns the available space in a file system.
+*
+* On `Windows`, this function uses the result of the Windows API function
+* $(LINK2 https://msdn.microsoft.com/en-us/library/windows/desktop/aa364937(v=vs.85).aspx, $(D GetDiskFreeSpaceExW)).
+*
+* On `POSIX` platforms, it uses the $(LINK2 http://man7.org/linux/man-pages/man3/statvfs.3.html,
+* $(D statvfs)) function.
+*
+* Due to inconsistent behaviour with the `statvfs` `Posix` function,
+* `FreeBSD` does not support `getAvailableDiskSpace`.
+*
+* Params:
+*     path = a valid path to any file in the file system
+*
+* Returns:
+*    ulong which contains the available space in bytes
+*
+* Throws:
+*     $(LREF FileException) in case statvfs (`POSIX`) or GetDiskFreeSpaceExW (`Windows`) fails.
+*/
+ulong getAvailableDiskSpace(string path) @trusted
+{
+    import std.exception : enforce;
+    import std.conv : text;
+
+    version (FreeBSD)
+    {
+        static assert(0, "Unsupported platform");
+    }
+    else version (Windows)
+    {
+        import core.sys.windows.winbase : GetDiskFreeSpaceExW, GetLastError;
+        import core.sys.windows.winnt : ULARGE_INTEGER;
+        import std.internal.cstring : tempCStringW;
+
+        ULARGE_INTEGER freeBytesAvailable;
+        auto err = GetDiskFreeSpaceExW(path.tempCStringW(), &freeBytesAvailable, null, null);
+        enforce(err != 0,
+            new FileException(text("Cannot get available disk space: ", GetLastError())));
+
+        return freeBytesAvailable.QuadPart;
+    }
+    else version (Posix)
+    {
+        import core.sys.posix.sys.statvfs : statvfs, statvfs_t;
+        import std.string : toStringz;
+
+        statvfs_t stats = void;
+        auto err = statvfs(path.toStringz(), &stats);
+        enforce(err == 0,
+            new FileException(text("Cannot get available disk space: ", err)));
+
+        return stats.f_bavail * stats.f_frsize;
+    }
+    else static assert(0, "Unsupported platform");
+}
+
+version (FreeBSD) {}
+else
+{
+    ///
+    @safe unittest
```

Comment:
```
version (FreeBSD) {} else
@safe unittest
```

is typically used for this. However, the documentation checker enforces that the examples on the documentation work on all platforms, so this needs to be moved in the test for now (or the documentation example checker needs to be adapted to support `version` blocks).